File: backend/app/models/public_authority.py
from sqlalchemy import Column, Integer, String, Text, Float, ForeignKey, DateTime, func
from sqlalchemy.orm import relationship

from ..database import Base


class PublicAuthority(Base):
    __tablename__ = "public_authorities"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(200), nullable=False)
    description = Column(Text, nullable=True)
    budget = Column(Float, default=0.0)
    address = Column(String(255), nullable=True)
    phone = Column(String(50), nullable=True)
    email = Column(String(150), nullable=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)

    user = relationship("User", back_populates="public_authority_profile")
    appels_offres = relationship("TenderCall", back_populates="authority")
    marches_publics = relationship("PublicContract", back_populates="authority")

File: frontend/src/types/publicAuthority.ts
export interface PublicAuthority {
  id: number;
  name: string;
  description?: string | null;
  budget?: number | null;
  address?: string | null;
  phone?: string | null;
  email?: string | null;
  user_id: number;
  created_at: string;
  updated_at: string;
}

export interface PublicAuthorityCreate {
  name: string;
  description?: string;
  budget?: number;
  address?: string;
  phone?: string;
  email?: string;
  user_id: number;
}

export type PublicAuthorityUpdate = Partial<Omit<PublicAuthorityCreate, "user_id">>;
